qcodes/instrument_drivers/QuTech/S5i.py

```diff
@@ -14,43 +14,49 @@ class S5i(Instrument):
 
     Args:
         name (str): name of the instrument.
-
         spi_rack (SPI_rack): instance of the SPI_rack class as defined in
             the spirack package. This class manages communication with the
             individual modules.
-
         module (int): module number as set on the hardware.
     """
 
     def __init__(self, name, spi_rack, module, **kwargs):
         super().__init__(name, **kwargs)
 
-        self.s5i = S5i_module(spi_rack, module)
+        self.s5i = S5i_module(spi_rack, module, frequency=41e6, enable_output=1,
+                              output_level=0)
 
-        self.add_parameter('use_external_reference',
-                           label='Use external reference',
-                           get_cmd=self._use_external_reference,
-                           set_cmd=self.s5i.use_external_reference,
-                           vals=Bool())
+        self.add_parameter('enabled',
+                           label='RF_on/off',
+                           set_cmd=self.s5i.enable_output_soft,
+                           vals=Bool(),
+                           docstring='Switched output on/off')
 
         self.add_parameter('frequency_stepsize',
                            label='Frequency stepsize',
                            get_cmd=self._get_stepsize,
                            set_cmd=self.s5i.set_stepsize,
                            units='Hz',
-                           vals=Numbers())
+                           vals=Numbers(),
```

Comment:
no limits here?

Reply:
There are, but bit complicated as it needs to be integer division of reference frequency and below 1024.
`reference_frequency / stepsize < 1024`

Reply:
It's checked in the underlying driver